kinto/views/admin.py

```diff
@@ -0,0 +1,168 @@
+        object_uris = list(object_uris_and_permissions.keys())
+        write_perm_principals = permission.get_objects_permissions(object_uris, ['write'])
+        to_delete = []
+        for object_uri, principals in zip(object_uris, write_perm_principals):
+            if 'write' not in principals:
+                continue
+            principals = principals['write']
+            # "Ownership" isn't a real concept in Kinto, so instead we
+            # define ownership as meaning "this user is the only one
+            # who can write to this object".
+            if principals == set([principal]):
+                to_delete.append(object_uri)
+
+        print("Going to delete", to_delete)
+        to_delete = condense_under_parents(self.request, to_delete)
+        print("Condensed into", to_delete)
+
+        # Group by (parent_uri, resource of child) to make fewer
+        # requests to storage backend.
+        objects_by_parent_uri = collections.defaultdict(list)
+        # We have to get the matchdict of the child here anyhow, so
+        # keep that to generate events later.
+        matchdicts_by_parent_uri = {}
+        for object_uri in to_delete:
+            parent_uri = get_parent_uri(object_uri)
+            resource_name, matchdict = core_utils.view_lookup(self.request,
+                                                              object_uri)
+            objects_by_parent_uri[(parent_uri, resource_name)].append(matchdict['id'])
+            # This overwrites previous matchdicts for the parent, but
+            # we'll only use the fields that are relevant to the
+            # parent, which will be the same for each child.
+            matchdicts_by_parent_uri[parent_uri] = matchdict
+
+        for (parent_uri, resource_name), object_ids in objects_by_parent_uri.items():
+            # Generate the parent matchdict from an arbitrary child's matchdict.
+            matchdict = {**matchdicts_by_parent_uri[parent_uri]}
+            matchdict.pop('id', None)
+
+            # FIXME: Deletes are paginated too.. ugh
+            batch_size = 100
+            for batch in slice_into_batches(object_ids, batch_size):
+                batch = list(batch)
+                filters = [Filter('id', batch, core_utils.COMPARISON.IN)]
+                records, _ = storage.get_all(
+                    collection_id=resource_name, parent_id=parent_uri,
+                    filters=filters
+                )
+                tombstones = storage.delete_all(
+                    collection_id=resource_name, parent_id=parent_uri,
+                    filters=filters
+                )
+                # FIXME: where do I get this from?
+                timestamp = None
+                notify_resource_event(self.request, parent_uri, timestamp,
+                                      tombstones, ACTIONS.DELETE, old=records,
+                                      resource_name=resource_name, resource_data=matchdict
+                )
```

Comment:
I wouldn't mind skipping resource events if that simplifies the code

Reply:
Getting resource events right is pretty important. For example, it's how the quota plugin handles adjusting quotas.